x/dispensation/types/records.go

```diff
@@ -59,12 +72,14 @@ func GetDistributionStatus(status string) DistributionStatus {
 		return DistributionStatus_DISTRIBUTION_STATUS_COMPLETED
 	case "Pending":
 		return DistributionStatus_DISTRIBUTION_STATUS_PENDING
+	case "Failed":
+		return DistributionStatus_DISTRIBUTION_STATUS_FAILED
 	default:
 		return DistributionStatus_DISTRIBUTION_STATUS_UNSPECIFIED
```

Comment:
Can consider returning a bool here as is done with GetClaimType

Reply:
I am still returning Pending records , if the status in unspecified (or anything other than the required ones ) . I can change it to return error to the  query if the status does not match the three prefixes specified

Reply:
changed it to return error if user queries with unknown status